src/community/oseo/oseo-service/src/main/java/org/geoserver/opensearch/eo/response/TemplatesProcessor.java

```diff
@@ -69,6 +116,78 @@ public String processTemplate(String collection, String templateName, Feature fe
         return sw.toString();
     }
 
+    public String processTemplate(SearchResults searchResults)
```

Comment:
This method code is made convoluted by the necessity to inspect the first feature to find out the collection name (and to some extent to see if it's a product or collection case, although that could have been identified looking at the request), and thus look up the right templates for it.

As a result, instead of having a linear header/content/footer sequence, we have the header handling mixed into the content loop. A simple way to avoid this is to use the PushBackFeatureIterator, which allows to read one feature, and push it back into the iterator... you can get the first, process the header, and then push it back, and process the contents, avoiding all the header logic mixup with the content logic.

Reply:
Moved header logic to the start of the function.